3_summarize/src/calculate_rmse.R

```diff
@@ -0,0 +1,38 @@
+
+calculate_rmse <- function(target_name, actual_fn, predicted_fn) {
+  
+  # Setup data to be able to merge and retain columns
+  
+  actual_df <- read_csv(actual_fn, col_types = cols()) %>% 
+    select(site_id, date, 
+           obs_opti_hab = opti_hab, 
+           obs_therm_hab = therm_hab,
+           obs_opti_therm_hab = opti_therm_hab)
+  
+  predicted_df <- read_csv(predicted_fn, col_types = cols()) %>% 
+    select(site_id, date, 
+           pred_opti_hab = opti_hab, 
+           pred_therm_hab = therm_hab,
+           pred_opti_therm_hab = opti_therm_hab)
+  
+  # Join the data and then calculate rmse per habitat for each lake
+  
+  actual_df %>% 
+    left_join(predicted_df, by = c("site_id", "date")) %>% 
+    group_by(site_id, date) %>% 
```

Comment:
right now you're comparing 1 pred to 1 obs for each RMSE calc, so not gaining a lot (this is the same as the subtraction). but i'll play with this next, as we planned, so I'm comfortable with this being temporary. Nice to have the file-reading in here anyway.